cgfs.py

```diff
@@ -332,21 +463,27 @@ def getattr(self, path, fh=None):
         parts = self.split_path(path)
         file = self.get_file(parts)
 
-        if file.stat is None and len(parts) > 3:
-            submission = self.get_submission(path)
+        if isinstance(file, TempFile):
+            return file.getattr()
 
+        if file.stat is None and len(parts) > 3:
             try:
-                query_path = submission.tld + '/' + '/'.join(parts[3:])
+                submission = self.get_submission(path)
             except CGAPIException as e:
                 handle_cgapi_exception(e)
 
+            query_path = submission.tld + '/' + '/'.join(parts[3:])
+
             if isinstance(file, Directory):
                 query_path += '/'
         else:
             submission = None
             query_path = None
 
-        return file.getattr(submission, query_path)
+        attrs = file.getattr(submission, query_path)
+        if self.fixed and isinstance(file, File):
+            attrs['st_mode'] &= ~0o222
+        return attrs
 
     # TODO?: Add xattr support
     def getxattr(self, path, name, position=0):
```

Comment:
Unrelated to this PR, but I think we can safely remove the `*xattr` methods if we're not going to use them.